### PR TITLE
Feature/everest 9862 secure field

### PR DESCRIPTION
**Pull request by elvismetaphor**
Create `SecureField` component

#### Demo Video:
SecureFieldExample_20201224

### CODE REVIEW

File: Sources/AltSwiftUI/Source/Views/Controls/SecureField.swift [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+//  SecureField.swift
+//  AltSwiftUI
+//
+//  Created by Elvis Lin on 2020/12/1.
+//
+
+import UIKit
+
+/// A view that allows text input in one line.
+public struct SecureField: View {
```
Comment (kevinwl02) [inline Sources/AltSwiftUI/Source/Views/Controls/SecureField.swift:11]: Currently, our TextField view can be setup to show secure entry. Did you consider wrapping the TextField view instead and configure it to show secure entry in the SecureField's body? That will reduce duplication between the 2 components and allow for easier scaling.

**Review by kevinwl02 — CHANGES_REQUESTED** [review]

File: Sources/AltSwiftUI/Source/Views/Controls/SecureField.swift (hunk 1 above)
Comment (elvismetaphor) [reply Sources/AltSwiftUI/Source/Views/Controls/SecureField.swift:11]: Thanks for your suggestion, and the pull request has been updated.

File: Sources/AltSwiftUI/Source/Views/Controls/SecureField.swift [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    public var viewStore = ViewValues()
+
+    let title: String
+    let onCommit: () -> Void
+    var text: Binding<String>
+    var isFirstResponder: Binding<Bool>?
+
+    public var body: View {
+        TextField(title, text: text, onCommit: onCommit)
```
Comment (kevinwl02) [inline Sources/AltSwiftUI/Source/Views/Controls/SecureField.swift]: 1. You need to set secure entry to true in the initializer
2. You need to pass the value of `isFirstResponder` to this view through a modifier

**Review by kevinwl02 — CHANGES_REQUESTED** [review]

File: Sources/AltSwiftUI/Source/Views/Controls/SecureField.swift (hunk 2 above)
Comment (elvismetaphor) [reply Sources/AltSwiftUI/Source/Views/Controls/SecureField.swift]: Thanks for your suggestion. I only keep a TextField inside SecureField now.

File: Sources/AltSwiftUI/Source/Views/Controls/SecureField.swift [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    /// Sets if this view is the first responder or not.
+    ///
+    /// Setting a value of `true` will make this view become the first
+    /// responder if not already.
+    /// Setting a value of `false` will make this view resign beign first
+    /// responder if it is the first responder.
+    ///
+    /// - important: Not SwiftUI compatible.
+    public func firstResponder(_ firstResponder: Binding<Bool>) -> Self {
+        _ = textField
```
Comment (kevinwl02) [inline Sources/AltSwiftUI/Source/Views/Controls/SecureField.swift]: You are just creating a new textField and not assign it anywhere. You need to create a new instance of self that is mutated with the new value of textField.
Example:
```
var view = self
view.textField = textField.firstResponder(firstResponder)
return view
```

**Review by kevinwl02 — CHANGES_REQUESTED** [review]

**Comment by kevinwl02** [comment]
Please add an implementation example in the example project and test the component.

**Review by kevinwl02 — APPROVED** [review]

**Pull request merged**